Tests iOS/ReaderTests.swift

```diff
@@ -0,0 +1,154 @@
+
+import XCTest
+import Sails
+import Combine
+import NIO
+
+class ReaderTests: XCTestCase {
+
+    var server: Application!
+    var app: PocketAppElement!
+
+    override func setUpWithError() throws {
+        continueAfterFailure = false
+
+        let uiApp = XCUIApplication()
+        app = PocketAppElement(app: uiApp)
+
+        server = Application()
+
+        server.routes.post("/graphql") { request, _ in
+            let apiRequest = ClientAPIRequest(request)
+
+            if apiRequest.isForSlateLineup {
+                return Response.slateLineup()
+            } else if apiRequest.isForSavesContent {
+                return Response.saves()
+            } else if apiRequest.isForArchivedContent {
+                return Response.archivedContent()
+            } else if apiRequest.isToFavoriteAnItem {
+                return Response.favorite()
+            } else if apiRequest.isToUnfavoriteAnItem {
+                return Response.unfavorite()
+            } else if apiRequest.isForTags {
+                return Response.emptyTags()
+            } else {
+                fatalError("Unexpected request")
+            }
+        }
+
+        try server.start()
+    }
+
+    override func tearDownWithError() throws {
+        try server.stop()
+        app.terminate()
+    }
+
+    func test_tappingSaves_dismissesReader_andShowsSaves() {
+        launchApp_andOpenItem()
+        app.readerView.savesBackButton.tap()
+        app.saves.wait()
+    }
+    
+    func test_archivingItem_dismissesReader_andShowsSaves() {
+        launchApp_andOpenItem()
+        server.routes.post("/graphql") { request, loop in
+            let apiRequest = ClientAPIRequest(request)
+            XCTAssertTrue(apiRequest.isToArchiveAnItem)
+            return Response.archive()
+        }
+        app.readerView.archiveButton.tap()
+        app.saves.wait()
+    }
+    
+    func test_tappingOverflowMenu_showsOverflowOptions() {
+        launchApp_andOpenItem()
+        openReaderOverflowMenu()
+        XCTAssertTrue(app.readerView.displaySettingsButton.exists)
+        XCTAssertTrue(app.readerView.favoriteButton.exists)
+        XCTAssertTrue(app.readerView.addTagsButton.exists)
+        XCTAssertTrue(app.readerView.deleteButton.exists)
+        XCTAssertTrue(app.readerView.shareButton.exists)
+    }
+    
+    func test_tappingDisplaySettings_showsDisplaySettings() {
+        launchApp_andOpenItem()
+        openReaderOverflowMenu()
+        openDisplaySettings()
+        XCTAssertTrue(app.readerView.fontStepperIncreaseButton.exists)
+        XCTAssertTrue(app.readerView.fontStepperDecreaseButton.exists)
+        openFontMenu()
+        XCTAssertTrue(app.readerView.fontSelection(fontName: "Graphik LCG").exists)
+        XCTAssertTrue(app.readerView.fontSelection(fontName: "Blanco OSF").exists)
+    }
+    
+    func test_tappingWebViewButton_showsSafari() {
+        launchApp_andOpenItem()
+        tapSafariButton()
+        validateSafariOpens()
```

Comment:
yes, even though we can't click on the button, the test will fail if Safari doesn't load - and that's all we need to check anyway, pressing the button would have been a 'nice to have' for testing that Reader re-appears after it's clicked, but this will be good enough to get started.

Reply:
agreed!